tiledb/api/src/datatype/arrow.rs

```diff
@@ -46,169 +53,667 @@ impl ArrowPrimitiveTypeNative for f64 {
+                            CellValNum::Fixed(nz) => {
+                                match nz.checked_mul(len) {
+                                    None => return Res::None,
+                                    Some(nz) => CellValNum::Fixed(nz),
+                                }
+                            }
+                            CellValNum::Var => CellValNum::Var,
+                        };
+                        Res::Exact(item, cell_val_num)
+                    }
+                }
+            }
+        }
+        ADT::LargeUtf8 => {
+            /*
+             * NB: arrow checks for valid UTF-8 but tiledb does not.
+             * This is not an exact conversion for that reason
+             * because we cannot guarantee invertibility.
+             */
+            Res::Inexact(Datatype::StringUtf8, CellValNum::Var)
+        }
+        ADT::LargeList(ref item) => {
+            if item.is_nullable() {
+                // no way to represent null values within a cell
+                Res::None
+            } else if item.data_type().primitive_width().is_none() {
+                /*
+                 * probably there are some cases we can handle,
+                 * but let's omit for now
+                 */
+                Res::None
+            } else if let Some(exact_datatype) = item
+                .metadata()
+                .get(ARROW_FIELD_METADATA_KEY_TILEDB_EXACT_MATCH)
+                .and_then(|s| Datatype::from_string(s))
+            {
+                Res::Exact(exact_datatype, CellValNum::Var)
+            } else {
+                match from_arrow(item.data_type()) {
+                    Res::None => Res::None,
+                    Res::Inexact(item, CellValNum::Fixed(nz))
+                        if nz.get() == 1 =>
+                    {
+                        Res::Inexact(item, CellValNum::Var)
+                    }
+                    Res::Exact(item, CellValNum::Fixed(nz))
+                        if nz.get() == 1 =>
+                    {
+                        Res::Exact(item, CellValNum::Var)
+                    }
+                    _ => {
+                        /*
+                         * We probably *can* fill in more cases, but either:
+                         * 1) we need to do work to keep the fixed cell val num around, doable but
+                         *    why bother right now
+                         * 2) we need to keep multiple levels of offsets, not supported right now
+                         */
+                        Res::None
+                    }
+                }
+            }
+        }
+        ADT::Binary | ADT::Utf8 | ADT::List(_) => {
+            /* offsets are 64 bits, these types use 32-bit offsets */
+            Res::None
+        }
+        ADT::BinaryView
+        | ADT::Utf8View
+        | ADT::ListView(_)
+        | ADT::LargeListView(_) => {
+            /* not something we can represent */
```

Comment:
I would make this comment more specific as to the why we can't support them. Assuming we're on the same page as to the why, I'd word it as "Not something we can represent easily without re-allocating contiguous buffers and changing the schema." Or more generally "Can't be supported directly without reallocating" or whatever.